fe2-android/app/src/main/java/com/github/dedis/student20_pop/detail/fragments/event/creation/RollCallEventCreationFragment.java

```diff
@@ -9,146 +9,108 @@
 import android.view.ViewGroup;
 import android.widget.Button;
 import android.widget.EditText;
-import android.widget.Toast;
 import androidx.annotation.NonNull;
 import androidx.annotation.Nullable;
 import com.github.dedis.student20_pop.databinding.FragmentCreateRollCallEventBinding;
 import com.github.dedis.student20_pop.detail.LaoDetailActivity;
 import com.github.dedis.student20_pop.detail.LaoDetailViewModel;
-import com.github.dedis.student20_pop.model.event.EventType;
 import java.time.Instant;
 
 /** Fragment that shows up when user wants to create a Roll-Call Event */
 public final class RollCallEventCreationFragment extends AbstractEventCreationFragment {
 
-  public static final String TAG = RollCallEventCreationFragment.class.getSimpleName();
-
-  private FragmentCreateRollCallEventBinding mFragBinding;
-  private LaoDetailViewModel mLaoDetailViewModel;
-  private EditText rollCallTitleEditText;
-  private Button confirmButton;
-  private Button openButton;
-
-  private final TextWatcher confirmTextWatcher =
-      new TextWatcher() {
-        @Override
-        public void beforeTextChanged(CharSequence s, int start, int count, int after) {
-          // Nothing needed here
-        }
-
-        @Override
-        public void onTextChanged(CharSequence s, int start, int before, int count) {
-          String meetingTitle = rollCallTitleEditText.getText().toString().trim();
-          boolean areFieldsFilled =
-              !meetingTitle.isEmpty() && !getStartDate().isEmpty() && !getStartTime().isEmpty();
-          confirmButton.setEnabled(areFieldsFilled);
-          openButton.setEnabled(areFieldsFilled);
-        }
-
-        @Override
-        public void afterTextChanged(Editable s) {
-          // Nothing needed here
-        }
-      };
-
-  public static RollCallEventCreationFragment newInstance() {
-    return new RollCallEventCreationFragment();
-  }
-
-  @Nullable
-  @Override
-  public View onCreateView(
-      @NonNull LayoutInflater inflater,
-      @Nullable ViewGroup container,
-      @Nullable Bundle savedInstanceState) {
-
-    mFragBinding = FragmentCreateRollCallEventBinding.inflate(inflater, container, false);
-
-    mLaoDetailViewModel = LaoDetailActivity.obtainViewModel(getActivity());
-
-    setDateAndTimeView(mFragBinding.getRoot(), this, getFragmentManager());
-    addDateAndTimeListener(confirmTextWatcher);
-
-    rollCallTitleEditText = mFragBinding.rollCallTitleText;
-    openButton = mFragBinding.rollCallOpen;
-    confirmButton = mFragBinding.rollCallConfirm;
-
-    mFragBinding.setLifecycleOwner(getActivity());
-
-    return mFragBinding.getRoot();
-  }
-
-  @Override
-  public void onActivityCreated(@Nullable Bundle savedInstanceState) {
-    super.onActivityCreated(savedInstanceState);
-
-    setupConfirmButton();
-    setupOpenButton();
-    setupCancelButton();
```

Comment:
I see what you mean, I will add these. Regarding your comment above about trying to keep as much as possible the master version, I intentionally moved the code around to have a fragment structure consistent with the others because I found it very different. Is adding back these methods enough?

Reply:
Moving the code around is ok, but some methods and events from the view model are completely different